test/runtests.jl

```diff
@@ -21,6 +21,53 @@ function draw2(μ)
     return (x,y)
 end
 
+measures = [
+    # Chain(x -> Normal(μ=x), Normal(μ=0.0))
+    For(3) do j Normal(σ=j) end
+    For(2,3) do i,j Normal(i,j) end
+    # Likelihood
+    # Pointwise
+    Normal() ^ 3
+    Normal() ^ (2,3)
+    # SpikeMixture(Normal(), 2)
+    # Dirac(0.0) + Normal()
+    # transforms
+    3 * Normal()
+    Bernoulli(0.2)
+    Beta(2,3)
+    Binomial(10,0.3)
+    Cauchy()
+    Dirichlet(ones(3))
+    Exponential()
+    Gumbel()
+    # InverseGamma(2)
+    Laplace()
+    LKJCholesky(3,2.0)
+    Multinomial(n=10,p=[0.2,0.3,0.5])
+    # MvNormal
+    NegativeBinomial(5,0.2)
+    Normal(2,3)
+    Poisson(3.1)
+    StudentT(ν=2.1)    
+    Uniform()
+    # CountingMeasure(Float64)
+    Dirac(π)
+    Lebesgue(ℝ)
+    # TrivialMeasure()
+]
+
+@testset "testvalue" begin
+    for μ in measures
+        @test logdensity(μ, testvalue(μ)) isa Float64
```

Comment:
If this is a universal requirement, can you also document somewhere, perhaps on the definition of `testvalue` that it should produce a point from the sample space of the measure such that evaluating `logdensity` on it produces a `Float64`?

Also, if this is the beginning of a test suite that all user-implemented measures should follow, then perhaps you should create a `test_measure` function in the source code of this package so that dependents of this package can use that function, and for now, the tests will just include this line.

Reply:
It would be useful to test somehow that `logdensity` does not needlessly promote. e.g. if the measure's parameters have a real eltype of `Float32`, and so does the test value, then `logdensity` should not promote to `Float64`.

Reply:
> If this is a universal requirement, can you also document somewhere, perhaps on the definition of `testvalue` that it should produce a point from the sample space of the measure such that evaluating `logdensity` on it produces a `Float64`?

Interesting, I hadn't thought of it as a universal requirement. The point is that `testvalue` returns a value (no error) and the result can be evaluated with `logdensity`. But maybe it should be? Seems reasonable to ask that everything is `Float64` unless otherwise specified.

> Also, if this is the beginning of a test suite that all user-implemented measures should follow, then perhaps you should create a `test_measure` function in the source code of this package so that dependents of this package can use that function, and for now, the tests will just include this line.

Good idea, I'll add this.

> It would be useful to test somehow that `logdensity` does not needlessly promote. e.g. if the measure's parameters have a real eltype of `Float32`, and so does the test value, then `logdensity` should not promote to `Float64`.

Makes sense to me. Maybe `test_measure` chould have a type parameter, or loop through some floating-point types